Makefile

```diff
@@ -565,7 +565,19 @@ ifdef LOCAL_BUILD
 	$(error LOCAL_BUILD must not be set for a release)
 endif
 
-release-prep: var-require-all-GIT_PR_BRANCH_BASE-GIT_REPO_SLUG-VERSION-CALICO_VERSION-COMMON_VERSION-CALICO_ENTERPRISE_VERSION
+handle-milestone: hack/bin/gh var-require-all-VERSION-GITHUB_TOKEN
+	@gh extension install valeriobelli/gh-milestone
+	@echo "Checking milestone $(VERSION) exists"
+	@gh milestone list --query $(VERSION) --repo $(REPO) --state all --json title | \
+		jq 'if .[0] == [] then error("milestone for release does not exist") else empty end'
+	@echo "Checking $(VERSION) milestone has no open PRs"
+	@gh search prs --milestone $(VERSION) --state open --json id | \
+		jq 'if . == [] then empty else error("there are open PRs in milestone") end'
+	$(eval MILESTONE_NUMBER := $(shell gh milestone list --query $(VERSION) --repo $(REPO) --state closed --json number --jq '.[0].number'))
+	@echo "Closing $(VERSION) milestone #$(MILESTONE_NUMBER)"
```

Comment:
You've already got in the readme to close the Milestone, I don't think we should do that here also, I think here we should just check that the milestone is closed.

Reply:
updated to check that the milestone is closed